src/OpenSSL/crypto.py

```diff
@@ -1952,27 +2000,114 @@ def add_revoked(self, revoked):
             # TODO: This is untested.
             _raise_current_error()
 
-    def export(self, cert, key, type=FILETYPE_PEM, days=100,
-               digest=_UNSPECIFIED):
+    def get_issuer(self):
         """
-        Export a CRL as a string.
+        Get the CRL's issuer.
 
-        :param cert: The certificate used to sign the CRL.
-        :type cert: :py:class:`X509`
+        .. versionadded:: 0.17
 
-        :param key: The key used to sign the CRL.
-        :type key: :py:class:`PKey`
+        :return: :py:class:`X509Name`
+        """
+        _issuer = _lib.X509_NAME_dup(_lib.X509_CRL_get_issuer(self._crl))
+        _openssl_assert(_issuer != _ffi.NULL)
+        _issuer = _ffi.gc(_issuer, _lib.X509_NAME_free)
+        issuer = X509Name.__new__(X509Name)
+        issuer._name = _issuer
+        return issuer
 
-        :param type: The export format, either :py:data:`FILETYPE_PEM`,
-            :py:data:`FILETYPE_ASN1`, or :py:data:`FILETYPE_TEXT`.
+    def set_version(self, version):
+        """
+        Set the CRL version.
 
-        :param int days: The number of days until the next update of this CRL.
+        .. versionadded:: 0.17
+
+        :param int version: The version of the CRL.
+        :return: :py:const:`None`
+        """
+        _openssl_assert(_lib.X509_CRL_set_version(self._crl, version) != 0)
 
+    def _set_boundary_time(self, which, when):
+        return _set_asn1_time(which(self._crl), when)
+
+    def set_lastUpdate(self, when):
+        """
+        Set when the CRL was last updated.
+
+        The timestamp is formatted as an ASN.1 GENERALIZEDTIME::
+
+            YYYYMMDDhhmmssZ
+            YYYYMMDDhhmmss+hhmm
+            YYYYMMDDhhmmss-hhmm
+
+        .. versionadded:: 0.17
+
+        :param bytes when: A timestamp string.
+        :return: :py:const:`None`
+        """
+        return self._set_boundary_time(_lib.X509_CRL_get_lastUpdate, when)
+
+    def set_nextUpdate(self, when):
+        """
+        Set when the CRL will next be udpated.
+
+        The timestamp is formatted as an ASN.1 GENERALIZEDTIME::
+
+            YYYYMMDDhhmmssZ
+            YYYYMMDDhhmmss+hhmm
+            YYYYMMDDhhmmss-hhmm
+
+        .. versionadded:: 0.17
+
+        :param bytes when: A timestamp string.
+        :return: :py:const:`None`
+        """
+        return self._set_boundary_time(_lib.X509_CRL_get_nextUpdate, when)
+
+    def sign(self, issuer_cert, issuer_key, digest='sha1'):
```

Comment:
Since this is a new API and there's no backwards compatibility concern I'd prefer this to not have a default arg for `digest`.